add baseHref

paulopereira98 opened the pull request:
Allow publishing to a github page that's not named after the user/institution. ex: user.github.io/repoName

erickzanardo commented:
Hey, thanks for the contribution.

I am unsure if we should add this though, flutter has already added a parameter on its own command (as seen here https://github.com/flutter/flutter/pull/80519), so I don't think we need to implement this on the action.

What do you think?

paulopereira98 commented:
Hi again, 
I noticed that this feature requires a placeholder tag on the index.html file which was added on pull [#80519](https://github.com/flutter/flutter/pull/80519) and only came out on the release version 2.5.0 (9/8/2021).
For flutter projects greated on earlier versions this is not the case.
I added a note to the readme, renamed the parameter and used the native implementation.

If you feel this pull is useful, I would be glad to contribute to your repo.

erickzanardo commented on README.md:
````diff
@@ -47,16 +47,28 @@ If you need to change that, the `targetBranch` property can be used
 ```
 
+By default, the generated page only works on `User or Organization sites` ex:`user.github.io/`. 
+You can change that by specifying the `baseHref`, so the site will work on a `Project Site`, ex:`user.github.io/repoName`.
````
```suggestion
You can change that by specifying the `baseHref` argument, so the site will work on a `Project Site`, ex:`user.github.io/repoName`.
```

erickzanardo commented on README.md:
```diff
@@ -47,16 +47,28 @@ If you need to change that, the `targetBranch` property can be used
 
+By default, the generated page only works on `User or Organization sites` ex:`user.github.io/`. 
+You can change that by specifying the `baseHref`, so the site will work on a `Project Site`, ex:`user.github.io/repoName`.
+The parameter `baseHref` must start and end with a forward slash `"/"`.
```
```suggestion

The parameter `baseHref` must start and end with a forward slash `"/"`.
```

erickzanardo commented on README.md:
```diff
@@ -47,16 +47,28 @@ If you need to change that, the `targetBranch` property can be used
+By default, the generated page only works on `User or Organization sites` ex:`user.github.io/`. 
+You can change that by specifying the `baseHref`, so the site will work on a `Project Site`, ex:`user.github.io/repoName`.
+The parameter `baseHref` must start and end with a forward slash `"/"`.
+(For projects created on flutter release version `2.2.3` or earlier, please manually edit the file `web/index.html`, changing the line `<base href="/">` to `<base href="$FLUTTER_BASE_HREF">`)
```
```suggestion
(For projects created on flutter version earlier than `2.5.0`, please manually edit the file `web/index.html`, changing the line `<base href="/">` to `<base href="$FLUTTER_BASE_HREF">`)
```

erickzanardo reviewed (commented):
Great work! I left a few suggestions on the README changes, we can merge and release this once they are accepted.
